components/Proposals/Instructions/Decoded/DecodedInstruction.tsx

```diff
@@ -0,0 +1,77 @@
+import { Provider } from '@coral-xyz/anchor';
+import { Card } from '@mantine/core';
+import { Connection, TransactionInstruction } from '@solana/web3.js';
+import { useEffect, useState } from 'react';
+
+import { RawInstructionCard } from '@/components/Proposals/Instructions/Raw';
+import { DefaultDecodedInstructionCard } from '@/components/Proposals/Instructions/Decoded/DefaultDecodedInstruction';
+import { MemoProgramInstructionCard } from '@/components/Proposals/Instructions/Decoded/SPL/Memo';
+import { TokenProgramInstructionCard } from '@/components/Proposals/Instructions/Decoded/SPL/Token';
+import {
+  InstructionDecoder,
+  SPL_MEMO_PROGRAM_ID,
+  SPL_TOKEN_PROGRAM_ID,
+} from '@/lib/instructions/coder';
+import { DecodedInstruction } from '@/lib/instructions/coder/types';
+
+const isDecodedInstruction = (
+  ix: TransactionInstruction | DecodedInstruction | undefined,
+): ix is DecodedInstruction => {
+  if (!ix) return false;
+  return (ix as DecodedInstruction).name !== undefined;
+};
+
+export const DecodedInstructionCard = ({
+  instruction,
+  provider,
+}: {
+  instruction: TransactionInstruction;
+  provider: Provider;
+}) => {
+  const [possiblyDecodedInstruction, setPossiblyDecodedInstruction] = useState<
+    DecodedInstruction | TransactionInstruction
+  >(instruction);
+
+  useEffect(() => {
```

Comment:
The thinking we have now is react-query to limit the requests.

Reply:
oh nice, react-query makes a lot of sense. i figured there had to be a better way than hammering requests every time these components are rendered. will take a look at other uses and make updates here.

thanks for pointing this out!

Reply:
@LukasDeco my understanding is we don't need use effect with useQuery AFAIK?

Reply:
that's basically right @R-K-H. Generally we should just do:
```
const {data: myState} = useQuery({...})
```

rather than:
```
const [myState, setMyState] = useState();
useEffect(() => {
asyncWork().then(res => setMyState(res));
}
```

Since it allows us to do it in less code making the components much cleaner, and of course at the same time adding the caching. Sometimes you may need some extra logic and controls and in that case a `useState` and `useEffect` may work better. But in general the simple cases just need a `useQuery` for sure.

Reply:
awesome, makes sense. thanks for the explainer. updating for the next revision!